connecting-to-xdai/run-own-node.md

```diff
@@ -0,0 +1,127 @@
+      > If your sync is very slow (extended beyond two days) then very likely your setup cannot catch up with the chain progress.
+    * ![](../.gitbook/assets/xdai-node/droplet.png)
+
+2. Remotely [connect](https://docs.digitalocean.com/products/droplets/how-to/connect-with-ssh/) to the Droplet using ssh. Probably something like `ssh root@157.245.5.42`
+* You should see the following on success:
+* ![](../.gitbook/assets/xdai-node/login.png)
+
+### 2. Install Nethermind and dependencies
+
+*Note: these commands are verbatim from [here](https://docs.nethermind.io/nethermind/ethereum-client/running-nethermind/running-the-client)*
+
+
+1. `wget [NETHERMIND_URL]` - use the most recent linux amd64 release from [downloads](https://downloads.nethermind.io/))  Probably something like `wget https://nethdev.blob.core.windows.net/builds/nethermind-linux-amd64-1.11.3-fe90423.zip`
+2. ```sudo apt-get update && sudo apt-get install libsnappy-dev libc6-dev libc6 unzip -y``` - Linux dependencies
+3. `unzip NETHERMIND_URL -d nethermind` - unzip to nethermind folder
+
+At this point, I diverged from the basic installation. This is because there are some specific configuration settings that are nice for running a node (and playing Dark Forest). 
+
+### 3. Configure Nethermind for Dark Forest
+
+So in the `nethermind` folder you will see `Nethermind.Launcher` and `Nethermind.Runner`.
+
+* ![](../.gitbook/assets/xdai-node/nethermind.png)
+
+From [ethhub](https://docs.ethhub.io/using-ethereum/ethereum-clients/nethermind/)
+
+> Nethermind.Launcher is an interactive prompt that allows you to choose the network which then calls Nethermind.Runner with the appropriate config file. This is the quickest way to get up and running.
+> Nethermind.Runner is the actual client binary similar to the geth binary and supports both cli options or a JSON configuration file. `Nethermind.Runner --config <path_to_config>`
+
+I added some custom configurations, so I choose to use the `Nethermind.Runner` binary with my custom config file.
+
+1. `cd nethermind` 
+  * We want to sync an xDai node, so here are the options for xDai (xdai_dfdao is my custom config): 
+  * ![](../.gitbook/assets/xdai-node/config.png)
+  * I choose to copy `xdai_pruned_mev.cfg` because:
+    1. I want the experiment with [mev](https://www.xdaichain.com/for-developers/developer-resources/mev-and-flashbots) on xDai
+    2. I want to save size by pruning, at the cost of historical queries:  
+        > pruning is the process of removing some of the intermediary state nodes - it saves some disk space but makes most of the historical state queries fail.
+
+2. `cp configs/CONFIG_FILE configs/YOUR_CONFIGNAME`
+  * for me, this was `cp configs/xdai_pruned_mev.cfg configs/xdai_dfdao.cfg`
+
+Edit your config file with something like `nano configs/xdai_dfdao.cfg` and change the following fields to allow access to your node from the internet:
+1. `JSONRPC: {"Enabled": true}`
```

Comment:
Did you try it all separate like that in your config file? Does it work?
I think wed want to collapse these into one. Its kind funky because were not adding the close bracket but I dont want them to wipe out any pre-existing configs...

```
JSONRPC: { 
  "Enabled": true,
  "Host": DIGITAL_OCEAN_PUBLIC_IP,
  "EthModuleConcurrentInstances": 6,
```

Reply:
```
  "JsonRpc": {
    "Enabled": true,
    "Timeout": 20000,
    "Host": "IP ADDRESS",
    "Port": 8545,
    "WebSocketsPort": 8546,
    "EthModuleConcurrentInstances": 6
  }
  ```